src/SourceBrowser/src/SourceIndexServer/Startup.cs

```diff
@@ -64,12 +65,14 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             app.UseStaticFiles();
             app.UseRouting();
 
-
             app.UseEndpoints(endPoints =>
             {
                 endPoints.MapRazorPages();
                 endPoints.MapControllers();
             });
+
+            // Retrieve and store the logger
+            Program.Logger = app.ApplicationServices.GetService<ILogger<Program>>();

```

Comment:
I don't think you are using this so we can probably remove this line, and in any case I don't think this is the right pattern when using loggers. Instead, You'd want to use the Logger Factory and DI to automatically inject different loggers into different services, like `ILogger<Startup>` so that logs can be filtered and parsed more easily. As I suggested in an earlier comment, this can be handled as a follow up.